src/mito2/src/sst/index.rs

```diff
@@ -117,31 +143,28 @@ pub(crate) struct IndexerBuilder<'a> {
     pub(crate) intermediate_manager: IntermediateManager,
     pub(crate) index_options: IndexOptions,
     pub(crate) inverted_index_config: InvertedIndexConfig,
+    pub(crate) fulltext_index_config: FulltextIndexConfig,
 }
 
 impl<'a> IndexerBuilder<'a> {
     /// Sanity check for arguments and create a new [Indexer] if arguments are valid.
     pub(crate) async fn build(self) -> Indexer {
         let mut indexer = Indexer {
             file_id: self.file_id,
+            file_path: self.file_path.clone(),
             region_id: self.metadata.region_id,
             last_memory_usage: 0,
 
             ..Default::default()
         };
 
         indexer.inverted_indexer = self.build_inverted_indexer();
-        if indexer.inverted_indexer.is_none() {
-            indexer.abort().await;
-            return Indexer::default();
-        }
-
-        indexer.puffin_writer = self.build_puffin_writer().await;
-        if indexer.puffin_writer.is_none() {
-            indexer.abort().await;
+        indexer.fulltext_indexer = self.build_fulltext_indexer().await;
+        if indexer.inverted_indexer.is_none() && indexer.fulltext_indexer.is_none() {
```

Comment:
Does the `indexer` need to abort before returning just like before?

Reply:
Agreed, although this `abort` doesn't do much, it reduces the mental burden for the code reader, so I will add it.